tap_learndash/streams.py

```diff
@@ -1,46 +1,755 @@
+    schema = th.PropertiesList(
+        th.Property("id", th.IntegerType),
+        th.Property("date", th.DateTimeType),
+        th.Property("date_gmt", th.DateTimeType),
+        th.Property("guid", th.ObjectType(
+            th.Property("rendered", th.StringType)
+        )),
+        th.Property("modified", th.DateTimeType),
+        th.Property("modified_gmt", th.DateTimeType),
+        th.Property("slug", th.StringType),
+        th.Property("status", th.StringType),
+        th.Property("type", th.StringType),
+        th.Property("link", th.StringType),
+        th.Property("title", th.ObjectType(
+            th.Property("rendered", th.StringType)
+        )),
+        th.Property("content", th.ObjectType(
+            th.Property("protected", th.BooleanType),
+            th.Property("rendered", th.StringType)
+        )),
+        th.Property("author", th.IntegerType),
+        th.Property("featured_media", th.IntegerType),
+        th.Property("menu_order", th.IntegerType),
+        th.Property("template", th.StringType),
+        th.Property("ld_topic_category", th.ArrayType(th.StringType)),
+        th.Property("ld_topic_tag", th.ArrayType(th.StringType)),
+        th.Property("materials_enabled", th.BooleanType),
+        th.Property("materials", th.ObjectType(
+            th.Property("rendered", th.StringType)
+        )),
+        th.Property("video_enabled", th.BooleanType),
+        th.Property("video_url", th.StringType),
+        th.Property("video_shown", th.StringType),
+        th.Property("video_auto_complete", th.BooleanType),
+        th.Property("video_auto_complete_delay", th.IntegerType),
+        th.Property("video_show_complete_button", th.BooleanType),
+        th.Property("video_auto_start", th.BooleanType),
+        th.Property("video_show_controls", th.BooleanType),
+        th.Property("video_focus_pause", th.BooleanType),
+        th.Property("video_resume", th.BooleanType),
+        th.Property("assignment_upload_enabled", th.BooleanType),
+        th.Property("assignment_points_enabled", th.BooleanType),
+        th.Property("assignment_points_amount", th.IntegerType),
+        th.Property("assignment_auto_approve", th.BooleanType),
+        th.Property("assignment_deletion_enabled", th.BooleanType),
+        th.Property("forced_timer_enabled", th.BooleanType),
+        th.Property("forced_timer_amount", th.IntegerType),
+        th.Property("course", th.IntegerType),
+        th.Property("lesson", th.IntegerType),
+        th.Property("assignment_upload_limit_extensions", th.StringType),
+        th.Property("assignment_upload_limit_size", th.StringType),
+        th.Property("assignment_upload_limit_count", th.BooleanType)
+    ).to_dict()
+
+
+class UserCourseProgressStream(LearnDashStream):
+    """Define custom stream."""
+    name = "user_course_progress"
+    path = "/users/{user_id}/course-progress"
+    primary_keys = ["user_id", "course"]
+    parent_stream_type = CoursesUsersStream
```

Comment:
This is where the "natural parent" is the `users` stream, which is not available to us. But, if we run this for every course, we are going to be pulling the same user mutliple times -- once for every course. Ideally we would run this once per user, but are there any streams that would give us that?